river/tree/splitter/base_splitter.py

```diff
@@ -14,30 +21,32 @@ def __init__(self):
         super().__init__()
 
     @abstractmethod
-    def update(self, att_val, target_val, sample_weight) -> "AttributeObserver":
+    def update(
+        self, att_val, target_val: base.typing.Target, sample_weight: float
+    ) -> "Splitter":
         """Update statistics of this observer given an attribute value, its target value
         and the weight of the instance observed.
 
         Parameters
         ----------
         att_val
-            The value of the attribute.
+            The value of the monitored attribute.
         target_val
             The target value.
         sample_weight
             The weight of the instance.
         """
 
     @abstractmethod
-    def probability_of_attribute_value_given_class(self, att_val, target_val) -> float:
+    def cond_proba(self, att_val, class_val: base.typing.ClfTarget) -> float:
         """Get the probability for an attribute value given a class.
 
         Parameters
         ----------
         att_val
             The value of the attribute.
-        target_val
-            The target value.
+        class_val
```

Comment:
I suggest keeping it as `target_val`, for consistency with the `update` method.

Reply:
I decided to use `class_val` to make it clear that conditional probabilities only apply to the classification case.
What do you think? I am totally cool with both of them.

Reply:
I updated all the occurrences to `target_val` for consistency.